typescript/infra/config/environments/mainnet3/warp/configGetters/getEverclearConfig.ts

```
@@ -0,0 +1,341 @@
+    signaturesData.signatures[asset],
+  )) {
+    const originName = CHAIN_ID_TO_NAME[originChainId];
+    if (!originName) continue;
+    result[originName] = {};
+
+    for (const [destChainId, feeInfo] of Object.entries(
+      destinations as Record<string, any>,
+    )) {
+      const destName = CHAIN_ID_TO_NAME[destChainId];
+      if (!destName) continue;
+
+      result[originName][destName] = {
+        ...feeInfo,
+      };
+    }
+  }
+
+  return result;
+}
+
+const getConfigFromFeeData = (
+  asset: EverclearAsset,
+  routerConfig: ChainMap<RouterConfigWithoutOwner>,
+): ChainMap<HypTokenRouterConfig> => {
+  const feeDataByChain = fetchFeeData(asset);
+  const tokenAddresses =
+    asset === 'USDC'
+      ? usdcTokenAddresses
+      : asset === 'WETH'
+        ? wethTokenAddresses
+        : usdtTokenAddresses;
+
+  // USDT is not supported on ink chain
+  let originChains = [...EVERCLEAR_CHAINS];
+  if (asset === 'USDT') {
+    originChains = originChains.filter((chain) => chain !== 'ink');
+  }
+
+  return Object.fromEntries(
+    originChains.map((chain) => {
+      const owner = owners[chain];
+      assert(owner, `Owner not found for ${chain}`);
+
+      // Define output assets and fee parameters per destination using real signature data
+      const outputAssets = Object.fromEntries(
+        originChains
+          .filter((c) => c !== chain)
+          .map((destChain) => [
+            destChain,
+            addressToBytes32(tokenAddresses[destChain]),
+          ]),
+      );
+
+      const everclearFeeParams = Object.fromEntries(
+        originChains
+          .filter((c) => c !== chain)
+          .map((destChain) => {
+            const feeData = feeDataByChain[chain]?.[destChain];
+            return [
+              destChain,
+              {
+                fee: parseInt(feeData?.tokenFee || '0'),
+                deadline: parseInt(feeData?.deadline || '0'),
+                signature: feeData?.sig || '',
```

should we throw if theres are undefined?